File: TZPU-main/4/Server4/src/Services/registerService.ts
import * as http from 'http';

class RegisterService {
    private readonly discoveryServiceUrl: string;
    private readonly httpClient: http.ClientRequest;

    constructor(httpClient: http.ClientRequest, discoveryServiceUrl: string) {
        this.httpClient = httpClient;
        this.discoveryServiceUrl = discoveryServiceUrl;
    }

    async registerWithDiscoveryServiceAsync(): Promise<void> {
        const serviceAddress = 'http://localhost:8080';
        const serviceName = 'your-service-name';
        const payload = JSON.stringify({ ServiceName: serviceName, ServiceAddress: serviceAddress });

        const options: http.RequestOptions = {
            hostname: 'your-discovery-service-url',
            port: 80,
            path: '/register',
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': payload.length.toString(),
            },
        };

        return new Promise<void>((resolve, reject) => {
            const req = http.request(options, (res) => {
                let data = '';

                res.on('data', (chunk) => {
                    data += chunk;
                });

                res.on('end', () => {
                    if (res.statusCode === 200) {
                        console.log(`Successfully registered ${serviceName} at ${serviceAddress}`);
                        resolve();
                    } else {
                        console.log(`Failed to register ${serviceName}. Status code: ${res.statusCode}`);
                        reject(`Failed to register ${serviceName}. Status code: ${res.statusCode}`);
                    }
                });
            });

            req.on('error', (error) => {
                console.error(`An error occurred during registration: ${error.message}`);
                reject(`An error occurred during registration: ${error.message}`);
            });

            req.write(payload);
            req.end();
        });
    }
}

export default RegisterService;

File: TZPU-main/Notification/Services/RegisterService.cs
//using System;
//using System.Net.Http;
//using System.Threading.Tasks;
//using Microsoft.Extensions.Configuration;

//public class RegisterService
//{
//    private readonly HttpClient _httpClient;
//    private readonly IConfiguration _configuration;

//    public RegisterService(HttpClient httpClient, IConfiguration configuration)
//    {
//        _httpClient = httpClient;
//        _configuration = configuration;
//    }

//    public async Task RegisterWithDiscoveryServiceAsync()
//    {
//        var serviceAddress = "http://localhost:8080";
//        var serviceName = "your-service-name";
//        var payload = new { ServiceName = serviceName, ServiceAddress = serviceAddress };
//        try
//        {
//            var response = await _httpClient.PostAsync("http://localhost:5084", new HttpContent());

//            if (response.IsSuccessStatusCode)
//            {
//                Console.WriteLine($"Successfully registered {serviceName} at {serviceAddress}");
//            }
//            else
//            {
//                Console.WriteLine($"Failed to register {serviceName}. Status code: {response.StatusCode}");
//            }
//        }
//        catch (Exception ex)
//        {
//            Console.WriteLine($"An error occurred during registration: {ex.Message}");
//        }
//    }
//}
